### ISSUE TITLE
`MInt` syntax with no uses of `Int2MInt` crashes

### ISSUE THREAD

**Issue by Baltoli** [runtimeverification/llvm-backend#798]
Compiling this code crashes the code generator; if the commented line is un-commented then it will compile and run correctly.
```k
module TEST
  imports INT
  imports MINT

  syntax MInt{1}

  // rule _:Int => Int2MInt(0)::MInt{1}
endmodule
```

If the code is ill-formed for some reason when the rule is not present, we should emit a friendlier error message either in the frontend or in the backend.

**Comment by Baltoli** [comment]
This is a known bug; there's no reason it _shouldn't_ work but is very low priority because there's no good reason to declare and not use one of these sorts.

**Comment by Baltoli** [comment]
Though @dwightguth comments that it's a confusing error for new users trying to experiment with `MInt{...}`; we should try and fix this.
